Signal/src/view controllers/FullImageViewController.m

```diff
@@ -128,56 +159,106 @@ - (void)populateImageView:(UIImage *)image {
 }
 
 - (void)initializeGestureRecognizers {
-    self.doubleTap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(imageDoubleTapped:)];
-    self.doubleTap.numberOfTapsRequired = 2;
-
-    self.singleTap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(imageSingleTapped:)];
-    [self.singleTap requireGestureRecognizerToFail:self.doubleTap];
-
-    self.singleTap.delegate = self;
-    self.doubleTap.delegate = self;
+    UITapGestureRecognizer *singleTap = [[UITapGestureRecognizer alloc] initWithTarget:self
+                                                                                action:@selector(imageDismissGesture:)];
+    singleTap.delegate = self;
+    [self.view addGestureRecognizer:singleTap];
+    
+    UITapGestureRecognizer *doubleTap = [[UITapGestureRecognizer alloc] initWithTarget:self
+                                                                                action:@selector(imageDismissGesture:)];
+    doubleTap.numberOfTapsRequired = 2;
+    doubleTap.delegate = self;
+    [self.view addGestureRecognizer:doubleTap];
+    
+    // UISwipeGestureRecognizer supposedly supports multiple directions,
+    // but in practice it works better if you use a separate GR for each
+    // direction.
+    for (NSNumber *direction in @[
+                                  @(UISwipeGestureRecognizerDirectionRight),
+                                  @(UISwipeGestureRecognizerDirectionLeft),
+                                  @(UISwipeGestureRecognizerDirectionUp),
+                                  @(UISwipeGestureRecognizerDirectionDown),
+                                  ]) {
+        UISwipeGestureRecognizer *swipe = [[UISwipeGestureRecognizer alloc] initWithTarget:self
+                                                                                    action:@selector(imageDismissGesture:)];
+        swipe.direction = (UISwipeGestureRecognizerDirection) direction.integerValue;
+        swipe.delegate = self;
+        [self.view addGestureRecognizer:swipe];
+    }
 
-    [self.view addGestureRecognizer:self.singleTap];
-    [self.view addGestureRecognizer:self.doubleTap];
+    UILongPressGestureRecognizer *longPress = [[UILongPressGestureRecognizer alloc] initWithTarget:self
+                                                                                            action:@selector(longPressGesture:)];
+    longPress.delegate = self;
+    [self.view addGestureRecognizer:longPress];
 }
 
 #pragma mark - Gesture Recognizers
 
-- (void)imageDoubleTapped:(UITapGestureRecognizer *)doubleTap {
-    CGPoint tap          = [doubleTap locationInView:doubleTap.view];
-    CGPoint convertCoord = [self.scrollView convertPoint:tap fromView:doubleTap.view];
-    CGRect targetZoomRect;
-    UIEdgeInsets targetInsets;
+- (void)imageDismissGesture:(UIGestureRecognizer *)sender {
```

Comment:
I think clang-format isn't changing these because our .clang-format doesn't include a `BreakBeforeBraces` directive.  Not sure if we inherit one from `BasedOnStyle: WebKit`.  I've tried specifying `Stroustrup` (which I think is what you prefer), but it doesn't seem to be working.